### DISCUSSION TITLE
Temperature dependent tensile/compressive properties in MatEditor

### DISCUSSION THREAD

**Discussion by admirMakas** [WelSimLLC/WelSim-Apps#23] [General]
Hello,

I was wondering if there is a way to store temperature dependent tensile/compressive properties for materials. In my limited evaluation I have not been able to find a way.

Thanks,
Admir

**Comment by lcheng9** [comment]
Hi Admir, Thank you for your post and question. You can use the test data to store the temperature-dependent strain-stress data, as shown in the figure below. 
![mateditor_test_data](https://github.com/WelSimLLC/WelSim-Apps/assets/106984354/73ceb654-528e-4b1f-8750-40d59deae554)

**Reply by admirMakas** [reply]
Hello,

Thanks for the quick response. I am already using test data to store the temperature dependent stress-strain curves. I was more looking for a way to store a table of yield strength and ultimate strength values as a function of temperature. I know that these values are already found in the stress-strain curves. But it would be beneficial if they can be explicitly stored. At the moment the application only allows one value of yield/ultimate strength to be stored under the "Life and Fatigue" tab.

**Comment by lcheng9** [comment]
Hello @admirMakas, thank you so much for your elaboration. I can understand your request now. The current yield and ultimate strength are constant and sufficient for most modeling cases. Do you mind sharing the purpose of using temperature-dependent yield/ultimate strength? Instead of posting here, you are also welcome to send the development request to info@welsim.com.

**Reply by admirMakas** [reply]
I do not mind posting here. In short short we do descent amount of transient thermo-mechanical simulation work that typically requires us to track how material properties/strengths change and evolve across time. To that end, being able to store CTE and elastic data (Young's Modulus, Poisson's Ratio, etc...) as a function of temperature would also be of great benefit. 

Maybe giving users the option of storing data in user-defined tables is the quickest solution. That way individuals can customize to their needs. 

Thanks,
Admir

**Comment by lcheng9** [comment]
@admirMakas, Thank you for sharing your thoughts about the new properties. Before we start to work on these new properties in MatEditor, may I confirm these design aspects with you?
1. A new material property type that supports temperature-time-stress tabular data.
2. Allow multiple properties to be presented and stored.
3. Allow the curve display for the multiple temperature sets. 
A picture of the GUI design is attached. 

![mateditor_design1](https://github.com/WelSimLLC/WelSim-Apps/assets/106984354/6493e085-f8c9-4900-a220-3d535dd014cf)

**Comment by lcheng9** [comment]
Hi @admirMakas, we added three new test data properties to the latest MatEditor, Uniaxial Plastic Strain Test Data, Shear Test Data - Viscoelastic, and Bulk Test Data- Viscoelastic, as shown in the figure below. Each property supports up to nine child test data sets, which should be sufficient for storing test data. You are welcome to try the latest version. I will be marking this one as the answer since the feature is developed. Thank you again for your suggestion. Please feel free to reopen this discussion or start a new post if you have any comments or concerns.

![mateditor_test_data_new](https://github.com/WelSimLLC/WelSim-Apps/assets/106984354/4d0fdacc-8c67-40ed-9aa7-7b6c270bbafd)
